### ISSUE TITLE
com.hazelcast.client.util.ClientStateListener doesn't work well with failover client

### ISSUE THREAD

**Issue by sumnerib** [hazelcast/hazelcast#18351]
There isn't really a nice way to use the provided `ClientStateListener` with the [Blue-Green Deployment and DR](https://docs.hazelcast.com/imdg/latest/clients/java.html#blue-green-deployment-and-disaster-recovery) client enterprise feature, due to the nature of the listener's [constructor](https://github.com/hazelcast/hazelcast/blob/167df41e272c8aa469aeca156900218591929be6/hazelcast/src/main/java/com/hazelcast/client/util/ClientStateListener.java#L57).

The following code snippet will cause an `InvalidConfigurationException`, since it violates the constraint that the failover configs match.

```
ClientFailoverConfig clientFailoverConfig = new ClientFailoverConfig();
clientFailoverConfig.addClientConfig(clientConfig1).addClientConfig(clientConfig2).setTryCount(3);
for (ClientConfig clientConfig : clientFailoverConfig.getClientConfigs()) {
    ClientStateListener clientStateListener = new ClientStateListener(clientConfig);
    clientStateListenerList.add(clientStateListener);
}
```

**Comment by sumnerib** [comment]
ZD 7136

**Comment by mmedenjak** [comment]
@hazelcast/apis is this planned to go into 5.0 or can we move it?

**Comment by sancar** [comment]
@mmedenjak I think we should fix it in 5.0.
